### PR TITLE
CB-14232 - support exact threshold, percentage base threshold and best

### PR DESCRIPTION
**Pull request by sodre90**
See detailed description in the commit message.

### CODE REVIEW

File: core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/model/DistroXScaleV1Request.java [hunk 1]
```diff
@@ -27,6 +28,10 @@
     @ApiModelProperty(value = InstanceGroupNetworkScaleModelDescription.NETWORK_SCALE_REQUEST)
     private NetworkScaleV1Request networkScaleRequest;
 
+    private AdjustmentType adjustmentType = AdjustmentType.EXACT;
+
+    private Long threshold;
+
```
Comment (lturcsanyi) [inline core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/model/DistroXScaleV1Request.java:34]: Is the missing @ApiModelProperty intentional?

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/autoscales/request/InstanceGroupAdjustmentV4Request.java [hunk 2]
```diff
@@ -25,6 +28,10 @@
     @ApiModelProperty(value = ModelDescriptions.InstanceGroupNetworkScaleModelDescription.NETWORK_SCALE_REQUEST)
     private NetworkScaleV4Request networkScaleRequest;
 
+    private AdjustmentType adjustmentType = AdjustmentType.EXACT;
+
+    private Long threshold;
+
```
Comment (lturcsanyi) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/autoscales/request/InstanceGroupAdjustmentV4Request.java:34]: Is the missing @ApiModelProperty intentional?

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/StackScaleV4Request.java [hunk 3]
```diff
@@ -30,6 +31,10 @@
     @ApiModelProperty(InstanceGroupAdjustmentModelDescription.FORCE)
     private Boolean forced;
 
+    private AdjustmentType adjustmentType = AdjustmentType.EXACT;
+
+    private Long threshold;
+
```
Comment (lturcsanyi) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/StackScaleV4Request.java:37]: Is the missing @ApiModelProperty intentional?

**Review by lturcsanyi — COMMENTED** [review]
Please fix commit message/PR description and please make sure there are useful comments around.

File: core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/model/DistroXScaleV1Request.java (hunk 1 above)
Comment (sodre90) [reply core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/model/DistroXScaleV1Request.java:34]: hmm.. I will add it, I think I missed it

**Comment by sodre90** [comment]
Added many log messages done! :)

**Review by lturcsanyi — APPROVED** [review]

**Pull request merged**